cpp/src/strings/regex/regex.cuh

```diff
@@ -34,7 +34,16 @@ namespace cudf {
 namespace strings {
 namespace detail {
 
-struct relist;
+/**
+ * @brief Template type used on `find` to specify desired position values in returned match_result
+ */
+enum class positional : int8_t {
+  BEGIN_END = 0,  /// both begin and end positions are returned
+  END_ONLY  = 1,  /// only the end position is returned
+};
+
+template <positional P>
+struct reljunk;
```

Comment:
I feel like I'm missing something about this name :D
Is it rel_junk? If yes, why junk, and if not, what does it stand for?

Reply:
Ah, I thought it might be _unknown_, but could not come up with any meaning for the lj part. Thanks!